src/lib/supabase/library/library.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  mockGetAllLibraryItems,
+  mockSearchLibraryItemsByAvailability,
+  mockSearchLibraryItemsByCategory,
+  mockSearchLibraryItemsByName,
+  mockSearchLibraryItemsByTags,
+  mockSearchAllLibraryItems,
+} from './mockData';
+import {
+  LibraryItem,
+  Hardware,
+  Book,
+  Film,
+  Controller,
+  Accessory,
+} from '~/models/models';
+
+//Stub Cloud Functions for the Library Page
+
+const getAllLibraryItems = (): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      resolve(mockGetAllLibraryItems);
+    }, 500);
+  });
+
+  return response;
+};
+
+const searchLibraryItemsByName = (
+  name: string
+): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      console.log('name', name);
```

Comment:
I typically recommend we leave console logs out of code on the branch.  If you find that you will be needing this often, it's okay to leave them, but maybe add a `TODO` comment above it to remove these later. (Same goes for any commented out code, usually like to get rid of it before merging to a target branch.)